python/metric_pipeline_tasks/PatchAnalysis.py

```diff
@@ -32,3 +30,16 @@ class PatchAnalysisTask(CatalogAnalysisBaseTask):
 
     ConfigClass = PatchAnalysisTaskConfig
     _DefaultName = "patchAnalysisTask"
+
+    def run(self, cat, vIds):
```

Comment:
I'm wondering if we should just put this in the `CatalogAnalysisBaseTask`.  Is there a reason not to?

Reply:
I didn't put it in `CatalogAnalysisBaseTask` because passing metadata (specifically, dataIds) doesn't make sense for the matched catalog tasks. I was trying to keep the run methods that require dataIds to be passed separate from those that don't. However, since we retarget the run methods in the analysis*.yaml pipelines, maybe it makes sense to have the more general version of the run method in `CatalogAnalysisBaseTask`?

Reply:
OK.  I don't really have a horse in this race, but do like to see duplication at a minimum.  I'll leave it up to you.

Reply:
While I agree with you that it would be much more tidy to have the new run and runQuantum methods in `CatalogAnalysisBaseTask`, I didn't implement that right now. There are subtle differences between `VisitAnalysis` and `PatchAnalysis` that made it seem more reasonable (to me) for them to have their own run methods.